File: c-sharp/recursion/ReverseString.cs
namespace c_sharp.recusrsion
{
    public class ReverseString
    {
        public static void Reverse(string s)
        {
            helper(0, s);
        }

        private static void helper(int index, string s)
        {
            if(s == null || index >= s.Length)
            {
                return;
            }

            helper(index + 1, s);
            
            System.Console.WriteLine(s[index]);

        }
    }
}

File: java/src/com/rohitsh/recursion/ReverseString.java
package com.rohitsh.recursion;

public class ReverseString {

    private static void helper(int index, String s){
        if(s == null || index >= s.length()) return;

        helper(index + 1, s);

        System.out.println(s.charAt(index));
    }

    public static void reverseString(String s) {
        helper(0, s);
    }

    public static void main(String[] args) {
        reverseString("hello world");
    }
}
